test/integration/puppetlabs/services/master/environment_transports_int_test.clj

```diff
@@ -0,0 +1,638 @@
+
+(use-fixtures :each
+              (fn [f]
+                (purge-env-dir)
+                (try
+                  (f)
+                  (finally
+                    (purge-env-dir)))))
+
+(defn get-env-transports
+  ([env-name]
+   (get-env-transports env-name nil))
+  ([env-name if-none-match]
+   (let [opts (if if-none-match
+                {:headers {"If-None-Match" if-none-match}})]
+     (try
+       (http-client/get
+        (str "https://localhost:8140/puppet/v3/"
+             "environment_transports?"
+             "environment="
+             env-name)
+        (merge
+         testutils/ssl-request-options
+         {:as :text}
+         opts))
+       (catch Exception e
+         (throw (Exception. "environment_transports http get failed" e)))))))
+
+(defn purge-all-env-caches
+  []
+  (http-client/delete
+   "https://localhost:8140/puppet-admin-api/v1/environment-cache"
+   testutils/ssl-request-options))
+
+(defn purge-env-cache
+  [env]
+  (http-client/delete
+   (str "https://localhost:8140/puppet-admin-api/v1/environment-cache?"
+        "environment="
+        env)
+   testutils/ssl-request-options))
+
+(defn response-etag
+  [request]
+  (get-in request [:headers "etag"]))
+
+(defn etag-with-gzip-suffix
+  [etag]
+  (if (.endsWith etag "--gzip")
+    etag
+    (str etag "--gzip")))
+
+(defn etag-without-gzip-suffix
+  [etag]
+  (str/replace etag #"--gzip$" ""))
+
+(defn response->map
+  [response]
+  (-> response :body (json/parse-string true)))
+
+
+(deftest ^:integration environment-transports-integration-cache-disabled-test
+  (testing "when environment cache is disabled for a transport request"
+    (bootstrap/with-puppetserver-running-with-config
+     app
+     (-> {:jruby-puppet {:gem-path gem-path
+                         :max-active-instances 1}}
+         (bootstrap/load-dev-config-with-overrides)
+         (ks/dissoc-in [:jruby-puppet
+                        :environment-class-cache-enabled]))
+     (let [foo-file (testutils/write-foo-pp-file
+                     "some kind of test data")
+           expected-response (json/encode mock-response)
+           response (get-env-transports "production")]
+       (testing "a successful status code is returned"
+         (is (= 200 (:status response))
+             (str
+              "unexpected status code for response, response: "
+              (ks/pprint-to-string response))))
+       (testing "no etag is returned"
+         (is (false? (contains? (:headers response) "etag"))))
+       (testing "the expected response body is returned"
+         (is (= expected-response
+                (response->map response))))))))
+
+
+(deftest ^:integration environment-transports-integration-cache-enabled-test1
+  (let [debug-log "./target/environment-transports-integration-cache-enabled.log"]
+    (fs/delete debug-log)
+    (bootstrap/with-puppetserver-running app
+     {:global {:logging-config
+               (str "./dev-resources/puppetlabs/services/"
```

Comment:
Are we just doing this logging switchover to make the tests easier to debug if something goes wrong? I don't see us actually looking at the resulting logs anywhere...

Reply:
Yeah, there's a try and then catch block where we print out the whole log. I actually triggered these failures in my refactor and didn't find that they added much value. In fact the log was so long that I couldn't see all of my failed tests. I'll remove this.